shell/plugins/steve/worker/web-worker.advanced.js

```diff
@@ -82,16 +83,41 @@ const removeFromWorkerQueue = (watchKey) => {
   });
 };
 
+/**
+ * Creates a resourceCache with the appropriate type
+ */
+
+const resourceCache = (type) => {
+  const CacheClass = cacheClasses[`${ capitalize(type) }Cache`] || cacheClasses.ResourceCache;
```

Comment:
There's a better way of doing this than just using capitalize, I could probably just camelcase all the cache classes and skip it but that just feels weird.

Reply:
changed